esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -2089,53 +2031,71 @@ void WsWuInfo::getWorkunitThorSlaveLog(const char *groupName, const char *ipAddr
     if (!rIO)
         throw MakeStringException(ECLWATCH_CANNOT_READ_FILE,"Cannot read file %s.",logName.str());
 
-    OwnedIFileIOStream ios = createBufferedIOStream(rIO);
+    Owned<IBufferedFileIOStream> ios = createBufferedFileIOStream(rIO);
     if (slaveNum > 0)
     {
-        StringBuffer line;
-        bool eof = false;
-        bool include = false;
+        readWorkunitLog(ios.get(), buf, outFile);
+    }
+    else
+    {//legacy wuid
+        appendIOStreamContent(buf, ios.get(), forDownload);
+    }
+}
+
+void WsWuInfo::readWorkunitLog(IBufferedFileIOStream* ios, MemoryBuffer& buf, const char* outFile)
```

Comment:
is 'buf' ever used?

Reply:
Not for downloading ZAP report. But, it may be used for viewing a file content.